clang/test/CIR/CodeGen/union-init.c

```diff
@@ -0,0 +1,42 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -emit-cir %s -o - | FileCheck %s
+
+typedef union {
+  int value;
+  struct {
+    int x : 16;
+    int y : 16;
+  };
+} A;
+
+void foo(int x) {
+  A a = {.x = x};
```

Comment:
I'd prefer if we could raise this initialization a bit, it shouldn't block progress though. Create an issue to track it.